cpp/src/parquet/properties.cc

```diff
@@ -26,10 +26,14 @@ namespace parquet {
 std::shared_ptr<ArrowInputStream> ReaderProperties::GetStream(
     std::shared_ptr<ArrowInputFile> source, int64_t start, int64_t num_bytes) {
   if (buffered_stream_enabled_) {
+    // ARROW-6180 / PARQUET-1636 Create thread-safe reader that references
+    // segment of source
+    std::shared_ptr<::arrow::io::InputStream> safe_stream;
+    PARQUET_THROW_NOT_OK(
+        ::arrow::io::RandomAccessFile::GetStream(source, start, num_bytes, &safe_stream));
     std::shared_ptr<::arrow::io::BufferedInputStream> stream;
-    PARQUET_THROW_NOT_OK(source->Seek(start));
     PARQUET_THROW_NOT_OK(::arrow::io::BufferedInputStream::Create(
```

Comment:
Have you evaluated whether `BufferedInputStream` helps performance? If Parquet reads large enough chunks that may not be the case.

Reply:
It's not a performance issue, it's a memory use issue. Column chunks in Parquet can be quite large, so if you instantiate all column readers at once you might be loading 1GB or more into memory. Databases find that to be unacceptable